thunder/executors/torchex.py

```diff
@@ -425,12 +425,77 @@ def _tensor_from_sequence_prims_transform(
     return tensor_from_sequence(seq_or_number, device=torch_device, dtype=torch_dtype)
 
 
+def _set_rng_state_prim_impl(new_state: torch.Tensor, device: torch.device | None) -> torch.Tensor:
+    if device is None:
+        device = torch.cuda.current_device()
+    torch.cuda.set_rng_state(new_state, device)
+    return new_state
+
+
+set_rng_state_prim_impl = ex.register_operator(
+    "set_rng_state_prim_impl",
+    meta=prims.set_rng_state.meta,
+    fn=_set_rng_state_prim_impl,
+    tags=(prims.OpTags.DONT_DCE,),
+)
+
+
+def _get_rng_state_prim_impl(state: torch.Tensor | None, device: torch.device | None) -> torch.Tensor:
+    if state is not None:
+        return state
+    if device is None:
+        return torch.cuda.get_rng_state()
+    return torch.cuda.get_rng_state(device)
+
+
+get_rng_state_prim_impl = ex.register_operator(
+    "get_rng_state_prim_impl",
+    meta=prims.get_rng_state.meta,
+    fn=_get_rng_state_prim_impl,
+)
+
+
+def _unpack_rng_state_prim_impl(s: torch.Tensor) -> tuple[int, int]:
+    seed, offset = torch.chunk(s, 2)
+    # We follow the nvFuser way here. The offset used by nvfuser = pytorch_offset // 4
+    # See Note [Divide offset by 4] https://github.com/NVIDIA/Fuser/blob/729f36c6fd0b86aa9d09a89ff3e406978cda13c7/csrc/rng.cpp#L54
+    return seed.view(torch.int64).item(), offset.view(torch.int64).item() // 4
+
+
+unpack_rng_state_prim_impl = ex.register_operator(
+    "unpack_rng_state_prim_impl",
+    meta=prims.unpack_rng_state.meta,
+    fn=_unpack_rng_state_prim_impl,
+)
+
+
+def _update_rng_state_prim_impl(seed: int, offset: int) -> torch.Tensor:
+    # We follow the nvFuser way here. pytorch_new_offset = (nvfuser_offset + 1) * 4
+    # See Note [Divide offset by 4] https://github.com/NVIDIA/Fuser/blob/729f36c6fd0b86aa9d09a89ff3e406978cda13c7/csrc/rng.cpp#L54
```

Comment:
nitpicking for a bit shorter line
```suggestion
    # See Note [Divide offset by 4] https://github.com/NVIDIA/Fuser/blob/729f36c/csrc/rng.cpp#L54
```